### DISCUSSION TITLE
Implementing Analytics Event Feature in Next.js TypeScript Tailwind CSS Firebase Project

### DISCUSSION THREAD

**Discussion by loyal812** [vercel/next.js#63789] [Help] [answered]
### Summary

Seeking guidance on integrating an analytics event feature into my Next.js project that uses TypeScript, Tailwind CSS, and Firebase. 

I need assistance in implementing analytics event tracking within my Next.js application to gather insights on user interactions and behavior.

I would appreciate any detailed instructions or resources on setting up analytics events in a Next.js project with TypeScript, Tailwind CSS, and Firebase.

### Additional information

```js
Next.js v14.1.0
```

**Comment by SouthLink** [comment]
next.js has a relevant example of what you said, please refer to the sample code below: 
[with-typescript](https://github.com/vercel/next.js/tree/canary/examples/with-typescript)
[with-tailwindcss](https://github.com/vercel/next.js/tree/canary/examples/with-tailwindcss)
[with-firebase](https://github.com/vercel/next.js/tree/canary/examples/with-firebase)

**Reply by loyal812** [reply]
Thank you for your interest in my post.
By the way, I would like to know the detailed steps to implement analytics event functionality using Firebase in next.js project.

**Reply by SouthLink** [reply]
https://firebase.google.com/codelabs/firebase-nextjs
This might help you

**Reply by loyal812** [reply]
I think you didn't understand what I meant. I want analytics event functionality in my next.js project.
I already implemented Firebase in my project.

**Comment by loyal812** [comment, marked as answer]
I made a mistake when trying to analyze events in my Next.js project using Firebase. All I had to do was integrate Firebase into my project, get the measurement ID, and voilà! After that, it was a breeze to add Google Analytics functionality, allowing me to easily review the analytics results on the Firebase event dashboard.

Check out my straightforward code below:

```jsx
import Script from 'next/script';

const GoogleAnalytics = ({ measurementId }: { measurementId: string }) => (
    <>
        <Script
            strategy="afterInteractive"
            src={`https://www.googletagmanager.com/gtag/js?id=${measurementId}`}
        />
        <Script id="google-analytics" strategy="afterInteractive">
            {`
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${measurementId}');
            `}
        </Script>
    </>
);

export default GoogleAnalytics;

**Reply by SouthLink** [reply]
The problem is that you are describing a problem that is not related to the code, this seems to be a function of the google tag manager and has nothing to do with firebase
